### PR TITLE
Fix tensorboard logging order and labels

### PR DESCRIPTION
**Pull request by gkielian**
1. Different labels for grad_norm and gns when doing tokens vs iterations, so they are in two separate graphs
2. Fixes the mfu and epoch ordering (needed to swap).
3. Allows for validation loss with the a single dataset for dataset_list mode

### CODE REVIEW

**Comment by gkielian** [comment]
Two things left to check:

1. with our new multidataset training changes, the single dataset mode now appears to have dips every evaluation interval in the training_loss -- this might have to do with tracking of the current iteration (this was slightly changed for multidataset training)
2. The grad norm has a bump at exactly 2000 iterations. This appears to affect only the grad norm and grad standard deviation, reducing the batch size does not seem to have an effect.

**Review by klei22 — APPROVED** [review]
